enhancements/network/egress-ip-multi-nic.md

```diff
@@ -0,0 +1,279 @@
+```
+
+Cluster manager would assign the `EgressIP` to worker 2.
+
+To allow packets to be steered into secondary networks, OVN reroute logical policies on `ovn_cluster_router` will redirect
+the packet out to the management port of the node which the egress IP is hosted.
+
+We will add rules in linux routing policy database control for each source Pod IP and lookup the correct routing table
+which will contain one route. This single route within a new managed routing table will direct the packet to the correct interface.
+
+We will also add a masquerade rule to iptables when a packet leaving a particular interface matches the source pod IP.
+
+
+See section `Design Details` for full implementation details.
+
+### User Stories
+As a user, I want the ability for cluster workload traffic to be routed to a network that is not managed by OVN and to have a configurable source IP.
+
+### Test Plan
+E2E tests verifying the correct source IP for deployment modes global IC and multi-zone IC:
+- Day 2 Networks CRUD including mask resizing
+- Configure new EgressIP both on ovn managed and unmanaged networks
+- Migrate existing EgressIP on ovn managed and unmanaged networks
+- Delete existing EgressIP on ovn managed and unmanaged networks
+- Add node
+- Delete node
+- Edit node
+- Add pod
+- Delete pod
+- Remove pod
+- Delete a valid rule manually 
+- Delete a valid IPtables rule manually
+- Delete a valid route manually
+- Delete a valid egress IP assigned to an interface manually
+- Valid egress service and egress IP functionality if both are active on the same k8 objs.
+Today, if egress service and egress IP both 'match' on the same pod, egress service is preferred.
+Continue this for this enhancement where egress service and egress IP overlap, prefer egress service.
+- Egress IP interface is now a child interface and attached to a bond which is now the IP of the EIP interface. Reconcile to move the EIP to the bond.
+- Egress IP interface is now a child interface and attached to a bond which is now a different IP of the original EIP interface. Reconcile to move the EIP
+to a different network and if none found, clear EIP node status and let cluster manager reconcile and find a new node.
+
+### API Extensions
+Add `RouteViaHost` field to `EgressIP` `.status`:
+
+```golang
+// The per node status, for those egress IPs who have been assigned.
+type EgressIPStatusItem struct {
+// Assigned node name
+Node string `json:"node"`
+// Assigned egress IP
+EgressIP string `json:"egressIP"`
+// Route via an OVN managed network or via a non OVN managed network
+RouteViaHost bool `json:"routeViaHost"`
+}
+```
+
+This will aid debugging of egress IP assignments but also aid egress IP controller located within ovnkube node to not
+have to recalculate if the egress IP is primary or secondary networks.
+
+### Risks and Mitigations
+- OVN kube node must recalculate all the matched pods for a given egress IP. This will require refactoring this logic
```

Comment:
I am confused with this sentence, cluster manager controller for egress IP doesn't care about pods?

Reply:
fixed.